festim/boundary_conditions/dirichlets/dc_imp.py

```diff
@@ -32,21 +36,47 @@ class ImplantationDirichlet(DirichletBC):
             assumed. Defaults to None.
         E_Kr (float, optional): recombination coefficient activation
             energy (eV). Defaults to None.
+        Kd_0 (float, optional): dissociation coefficient pre-exponential
+            factor (m-2 s-1 Pa-1). If None, instantaneous dissociation will be
+            assumed. Defaults to None.
+        E_Kd (float, optional): dissociation coefficient activation
+            energy (eV). Defaults to None.
+        P (float or sp.Expr, optional): partial pressure of H (Pa). Defaults to None.
     """
 
-    def __init__(self, surfaces, phi, R_p, D_0, E_D, Kr_0=None, E_Kr=None) -> None:
+    def __init__(
+        self,
+        surfaces,
+        phi,
+        R_p,
+        D_0,
+        E_D,
+        Kr_0=None,
+        E_Kr=None,
+        Kd_0=None,
+        E_Kd=None,
+        P=None,
+    ) -> None:
         super().__init__(surfaces, field=0, value=None)
         self.phi = phi
         self.R_p = R_p
         self.D_0 = D_0
         self.E_D = E_D
         self.Kr_0 = Kr_0
         self.E_Kr = E_Kr
+        self.Kd_0 = Kd_0
+        self.E_Kd = E_Kd
+        self.P = P
 
     def create_expression(self, T):
         phi = f.Expression(sp.printing.ccode(self.phi), t=0, degree=1)
         R_p = f.Expression(sp.printing.ccode(self.R_p), t=0, degree=1)
-        sub_expressions = [phi, R_p]
+        if self.P is not None:
+            P = f.Expression(sp.printing.ccode(self.P), t=0, degree=1)
+            sub_expressions = [phi, R_p, P]
```

Comment:
What you can do here instead of creating the sub_expressions list in both cases is to first create it with phi and Rp outside of the if statement and then if P is not None, append P to the list.

Reply:
True